src/OpenSearch/GuzzleHttpClientFactory.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSearch;
+
+use GuzzleHttp\Client as GuzzleClient;
+use GuzzleHttp\Exception\ConnectException;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Middleware;
+use Psr\Http\Client\ClientInterface;
+
+/**
+ * Builds an OpenSearch client using Guzzle.
+ */
+class GuzzleHttpClientFactory implements HttpClientFactoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function create(array $options): ClientInterface
+    {
+        if (!isset($options['base_uri'])) {
+            throw new \InvalidArgumentException('The base_uri option is required.');
+        }
+        // Set default configuration.
+        $defaults = [
+            'headers' => [
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+                'User-Agent' => sprintf('opensearch-php/%s (%s; PHP %s)', Client::VERSION, PHP_OS, PHP_VERSION),
+            ],
+        ];
+
+        $logger = $options['logger'] ?? null;
+        unset($options['logger']);
+
+        $maxRetries = $options['max_retries'] ?? 0;
+        unset($options['max_retries']);
+
+        // Merge the default options with the provided options.
+        $config = array_merge_recursive($defaults, $options);
+
+        $stack = HandlerStack::create();
```

Comment:
should we do class_exists check and throw a exception that the user should install package x using composer?

Reply:
I added Guzzle and Symfony HTTP Clients to the suggests. They will get an error if they try to use it with a missing class anyway.